src/modules/CrudMethods.js

```diff
@@ -0,0 +1,73 @@
+import Task from './task.js';
+
+class TaskList {
+  constructor() {
+    this.tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+    this.taskForm = document.getElementById('task-form');
+    this.taskInput = document.getElementById('task-input');
+    this.taskList = document.getElementById('task-list');
+    this.clearCompletedBtn = document.getElementById('clear-completed');
+    this.taskForm.addEventListener('submit', this.addTask.bind(this));
+    this.clearCompletedBtn.addEventListener('click', this.clearCompletedTasks.bind(this));
+    this.displayTasks();
+  }
+
+  addTask(e) {
+    e.preventDefault();
+    const taskName = this.taskInput.value.trim();
+    if (taskName !== '') {
+      const task = new Task(taskName, false);
+      this.tasks.push(task);
+      this.displayTasks();
+      this.taskInput.value = '';
+      localStorage.setItem('tasks', JSON.stringify(this.tasks));
+    }
+  }
+
+  removeTask(id) {
+    this.tasks = this.tasks.filter((task) => task.id !== id);
+    this.displayTasks();
+    localStorage.setItem('tasks', JSON.stringify(this.tasks));
+  }
+
+  toggleTaskStatus(id) {
+    const index = this.tasks.findIndex((task) => task.id === id);
+    this.tasks[index].status = !this.tasks[index].status;
+    this.displayTasks();
+    localStorage.setItem('tasks', JSON.stringify(this.tasks));
+  }
+
+  clearCompletedTasks() {
+    this.tasks = this.tasks.filter((task) => !task.status);
+    this.displayTasks();
+    localStorage.setItem('tasks', JSON.stringify(this.tasks));
+  }
+
+  resetTasks() {
+    this.tasks = [];
+    localStorage.removeItem('tasks');
+    this.displayTasks();
+  }
+
+  displayTasks() {
+    this.taskList.innerHTML = '';
+    this.tasks.forEach((task) => {
+      const taskElement = document.createElement('li');
+      taskElement.innerHTML = `
+        <input type='checkbox' ${task.status ? 'checked' : ''}>
+        <span>${task.name}</span>
+        <button class='delete-btn'>X</button>
+      `;
+      const checkbox = taskElement.querySelector('input[type=checkbox]');
+      checkbox.addEventListener('change', () => this.toggleTaskStatus(task.id));
+      const deleteBtn = taskElement.querySelector('.delete-btn');
+      deleteBtn.addEventListener('click', () => this.removeTask(task.id));
+      if (task.status) {
+        taskElement.querySelector('span').style.textDecoration = 'line-through';
+      }
+      this.taskList.appendChild(taskElement);
+    });
+  }
+}
+
+export default TaskList;
```

Comment:
- [ ] [REQUIRED] Hello  Khan, you quite have. an awesome UI design here 👍🏽 
However, it looks like you skipped the part of the project requirement that states that you need to add a function for editing task descriptions. Kindly implement this 🥇 

- [ ] [REQUIRED] Also, we need to ensure that the new tasks have the property `index` set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5) and let's also take note that ...
- [ ] [REQUIRED] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.

Let's fix these 💪🏽

Reply:
@Whoistolu I made all the required changes. Now I am going for a second review. Thanks.

Reply:
Awesome 👏 
@fpsapc